src/pdfxml.cpp

```diff
@@ -736,7 +740,7 @@ wxPdfTable::GetLastRowsOnPage() const
         //since m_bodyRowLast and m_headRowLast are alway one behind last, last row on page is always one behind too.
         lastRows.Add(row);
         //Maybe we have a header at the top of the next page
-        y = m_document->GetHeaderHeight();
+        y = topTotalMargin;
       }
       if (writeHeader)
```

Comment:
IMHO this `if` statement has to be removed. If I see it correctly the table header height is added for every table body row. This is wrong.

With the correction of `topTotalMargin` (see my first comment) the table header, page header, and page margin will be taken into account correctly.

BTW, in principle the table could be degenerated in that sense that the table header and the first table row are higher than the available space on the page. In that case we should probably issue at least a warning message.

Reply:
I made the suggested adjustments.

You're right, we should issue a warning in case that the table header + the first row doesn't fit on empty page. Currently `wxLOG_ERROR` is used for showing error message in wxpdfdoc, right?

Reply:
> I made the suggested adjustments.

Thanks. I already approved the changes, but I didn't merge them yet, in case you add a check for the corner case with header and first row higher than available page space.

> You're right, we should issue a warning in case that the table header + the first row doesn't fit on empty page. Currently `wxLOG_ERROR` is used for showing error message in wxpdfdoc, right?

Yes. Currently the function `wxLogError()` is used for reporting errors. Please let me know whether you intend to add a check in your PR. If not, I will merge your PR as is, because such an error should normally not occur at all.

Reply:
I added an error handling for this corner case to my PR. Thank you for your fast response to my PR.